graylog2-server/src/main/java/org/graylog2/streams/StreamServiceImpl.java

```diff
@@ -138,13 +138,18 @@ public Stream load(String id) throws NotFoundException {
         }
 
         List<DBObject> results = query(StreamImpl.class, query);
+        List<String> streamIds = results.stream()
+            .map(o -> o.get("_id").toString())
+            .collect(Collectors.toList());
+
+        Map<String, List<StreamRule>> allStreamRules = streamRuleService.loadForStreamIds(streamIds);
+
         for (DBObject o : results) {
             String id = o.get("_id").toString();
-            List<StreamRule> streamRules = null;
-            try {
-                streamRules = streamRuleService.loadForStreamId(id);
-            } catch (NotFoundException e) {
-                LOG.info("Exception while loading stream rules: " + e);
+
+            final List<StreamRule> streamRules = allStreamRules.getOrDefault(id, Lists.newArrayList());
```

Comment:
This should use `Collections.emptyList()` instead of `Lists.newArrayList()` to prevent the instantiation of unused objects.